polytope_server/common/observability/otel.py

```diff
@@ -0,0 +1,86 @@
+from opentelemetry import trace
+from opentelemetry.propagate import inject, extract
+from contextlib import contextmanager
+import logging
+
+def add_trace_context(request):
+    carrier = {}
+
+    # Inject the context into the carrier
+    inject(carrier)
+
+    # Adding trace_id and span_id to the request data for logging
+    request.otel_trace_ctx = {
+        'carrier': carrier  # Store the injected carrier
+    }
+
+    current_span = trace.get_current_span()
+    current_span.set_attribute("polytope.request.id", request.id)
```

Comment:
Why do we set this attribute in add_ and restore_? It seems like this should be done only in create_new_span and otherwise it seems that it will be unreliably set.

Reply:
We are passing the context between services, therefore, each service starts a new span where we restore the context (so we keep track of the propagation) but the attributes are missing and need to be updated again